ui/page/dcrdex/dex_onboarding_page.go

```diff
@@ -844,76 +892,207 @@ func (pg *DEXOnboarding) HandleUserInteractions() {
+						pm.SetLoading(false)
+						return false
+					}
 
+					nextStep()
+					pm.Dismiss()
+					pg.ParentWindow().Reload()
+					return true
+				})
+			pg.ParentWindow().ShowModal(dexPasswordModal)
 		case onboardingPostBond:
+			if pg.isLoading {
+				break
+			}
+
 			// Validate all input fields.
 			hasEnough := pg.bondAccountHasEnough()
 			bondStrengthOk := pg.validateBondStrength()
 			if !hasEnough || !bondStrengthOk {
-				return
+				break
+			}
+
+			asset := pg.bondSourceWalletSelector.SelectedWallet()
+			bondAsset := pg.bondServer.bondAssets[asset.GetAssetType()]
+			postBond := &core.PostBondForm{
+				Addr:      pg.bondServer.url,
+				AppPass:   pg.dexPass,
+				Asset:     &bondAsset.ID,
+				Bond:      uint64(pg.newTier) * bondAsset.Amt,
+				Cert:      pg.bondServer.cert,
+				FeeBuffer: pg.core.BondsFeeBuffer(bondAsset.ID),
 			}
 
-			// TODO: Post bond, wait for confirmations and redirect to market page.
-			pg.currentStep = onBoardingStepWaitForConfirmation
-			// Scroll to the top of the confirmation page after leaving the long
-			// post bond form.
-			pg.scrollContainer.Position.Offset = 0
+			pg.isLoading = true
+			pg.bondConfirmationInfo = new(bondConfirmationInfo)
+			go func() {
+				defer func() {
+					pg.isLoading = false
+				}()
+
+				res, err := pg.core.PostBond(postBond)
+				if err != nil {
+					pg.notifyError(fmt.Sprintf("Failed to post bond: %v", err))
+					return
+				}
+
+				pg.bondConfirmationInfo.requiredBondConf = res.ReqConfirms
+				pg.bondConfirmationInfo.bondCoinID = res.BondID
+				pg.currentStep = onBoardingStepWaitForConfirmation
+				pg.scrollContainer.Position.Offset = 0 // Scroll to the top of the confirmation page after leaving the long post bond form.
+				pg.ParentWindow().Reload()
+			}()
+
+			// TODO: handle onBoardingStepWaitForConfirmation and display market
+			// page when we have enough confirmation.
 		}
 	}
 }
 
+func (pg *DEXOnboarding) notifyError(errMsg string) {
+	errModal := modal.NewErrorModal(pg.Load, errMsg, modal.DefaultClickFunc())
+	pg.ParentWindow().ShowModal(errModal)
+}
+
 // bondAccountHasEnough checks if the selected bond account has enough to cover
 // the bond costs.
 func (pg *DEXOnboarding) bondAccountHasEnough() bool {
 	ac := pg.bondSourceAccountSelector.SelectedAccount()
-	if ac.Balance.Spendable.ToCoin() < float64(pg.newTier)*20 /* TODO: Use actual bond cost + reservations + fees */ {
-		pg.bondSourceAccountSelector.SetError(values.String(values.StrInsufficientFundsInAccount))
+	asset := pg.bondSourceWalletSelector.SelectedWallet()
+	bondAsset := pg.bondServer.bondAssets[asset.GetAssetType()]
+	bondsFeeBuffer := pg.core.BondsFeeBuffer(bondAsset.ID)
+	bondCost := uint64(pg.newTier)*bondAsset.Amt*bondOverlap + bondsFeeBuffer
+	bondAmt := asset.ToAmount(int64(bondCost))
+	if ac.Balance.Spendable.ToInt() < bondAmt.ToInt() {
+		pg.bondSourceAccountSelector.SetError(values.StringF(values.StrInsufficientBondAmount, bondAmt.String()))
```

Comment:
Core doesn't have a method to check this? I'd rather defer this to core, even if it means the user won't see an error message until core.PostBond errors for insufficient funds.

Reply:
> Core doesn't have a method to check this?

Core will just fail when it attempts to make payments but I don't think it's a good UX for users to fail before they know they can choose another wallet with sufficient funds.